internal/bminventory/inventory.go

```diff
@@ -3700,8 +3700,27 @@ func (b *bareMetalInventory) DownloadMinimalInitrd(ctx context.Context, params i
 	}
 
 	var netFiles []staticnetworkconfig.StaticNetworkConfigData
+	var scriptContent, serviceContent string
 	if infraEnv.StaticNetworkConfig != "" {
-		netFiles, err = b.staticNetworkConfig.GenerateStaticNetworkConfigData(ctx, infraEnv.StaticNetworkConfig)
+		// backward compatibility - nmstate.service has been available on RHCOS since version 4.13+, therefore, we should maintain both flows
+		ocpVersion, err := version.NewVersion(infraEnv.OpenshiftVersion)
+		if err != nil {
+			return common.GenerateErrorResponder(errors.Errorf("Failed to parse OCP version"))
+		}
+		minimalVersionForNmstatectl, err := version.NewVersion(constants.MinimalVersionForNmstatectl)
+		if err != nil {
+			return common.GenerateErrorResponder(errors.Errorf("Failed to parse minimal ocp version for nmstatectl"))
+		}
+		// TODO: Remove the architecture condition after fetching the nmstatectl binary from rootfs.
+		if ocpVersion.GreaterThanOrEqual(minimalVersionForNmstatectl) && infraEnv.CPUArchitecture == "x86_64" {
```

Comment:
We have a helper for this [here](https://github.com/openshift/assisted-service/blob/17927d8dfe54d1e5c0bde382f36278b2f9f91119/internal/common/version.go#L33)

Reply:
Also we have constants for the architectures [here](https://github.com/openshift/assisted-service/blob/17927d8dfe54d1e5c0bde382f36278b2f9f91119/internal/common/common.go#L44-L52)